server/commands/accountProvisioner.test.ts

```diff
@@ -193,4 +194,69 @@ describe("accountProvisioner", () => {
     const collectionCount = await Collection.count();
     expect(collectionCount).toEqual(1);
   });
+
+  it("should skip findExistingTeam in hosted context", async () => {
+    expect(false).toBeTruthy(); // fail on purpose
+  });
+
+  it("should create new team if no team in self-hosted context", async () => {
+    const env = process.env;
+    process.env = { ...env, DEPLOYMENT: "xxx" }; // self to something that isn't hosted
+
+    const existingTeam = await findExistingTeam({
+      name: "slack",
+      providerId: uuidv4(),
+    });
+    expect(existingTeam).toBeNull();
+
+    process.env = env; // reset
+  });
+
+  it("should return existing team, in self-hosted context", async () => {
+    const env = process.env;
+    process.env = { ...env, DEPLOYMENT: "xxx" }; // self to something that isn't hosted
+
+    const team = await buildTeam();
+    const authenticationProvider = (await team.getAuthenticationProviders())[0];
+    const existingTeam = await findExistingTeam({
```

Comment:
This file should be tests for accountProvisioner, we seem to be testing `findExistingTeam` here instead